src/actions/api.js

```diff
@@ -0,0 +1,111 @@
+import { RSAA } from 'redux-api-middleware';
+
+import * as types from './types';
+
+const HEADERS_BASE = {
+  Accept: 'application/json',
+  'Content-Type': 'application/json',
+};
+
+const API_URL_BASE = 'http://localhost:8000/';
+
+/*
+* Redux API Middleware (https://github.com/agraboso/redux-api-middleware)
+* will intercept any action that contains the RSAA key, and instead dispatch
+* some of the actions defined in the types property below, based on how
+* the request resolves. It uses the new JavaScript Fetch API internally.
+*/
+
+export const fetchUsers = () => {
+  return {
+    [RSAA]: {
+      endpoint: API_URL_BASE + 'users',
+      method: 'GET',
+      types: [
+        types.USERS_FETCH_REQUEST,
+        types.USERS_FETCH_SUCCESS,
+        types.USERS_FETCH_FAILURE,
+      ],
+      headers: HEADERS_BASE,
+    },
+  };
+};
+
+/*
+* This is more or less functionally equivalent with the commented out
+* redux-api-middleware fetchBooks action defined below. There are many
+* different ways to handle async actions in redux. redux-api-middleware is
+* a nice abstraction (IMO ;) for AJAX requests, because they always have
```

Comment:
Hah jeez. it IS a nice abstraction but just like any tool it isn't always the right choice ;)